geode-core/src/main/java/org/apache/geode/internal/cache/tier/sockets/ServerConnection.java

```diff
@@ -1231,6 +1231,7 @@ long putSubject(Subject subject, long existingUniqueId) {
       secureLogger.debug("update subject on client proxy {} with uniqueId {}", clientProxy,
           uniqueId);
       clientProxy.setSubject(subject);
```

Comment:
Feels like `setSubject` should have the sufficient notification framework to inform our waiting thread.

Reply:
since they use different lock, I will keep it in different methods.

Reply:
When would we ever mutate the subject and not want to notify any waiters?

Reply:
the `setSubject` is also called when registering a new client